### ISSUE TITLE
TNB Minecraft Integration

### ISSUE THREAD

**Issue by Kycermann** [thenewboston-developers/Projects#206]
# Project Details

#### How would you describe your project?
Players can deposit TNBC into their in-game accounts (and withdraw it), tip players (without paying fees) and receive TNBC in-game.

/tnb deposit
/tnb tip <PlayerName> [amount]
/tnb withdraw <address> <amount>

#### What problem does this project solve?
Integrates TNB with the Minecraft community.

#### Who is the target market?
Minecraft players and those in the TNB community who play Minecraft.

#### How will this project benefit the network and/or the community?
It'll expose people to TNB. Say you've never heard ot TNB and you're tipped some- now you're a TNB holder!

#### How is this project centered around thenewboston?
Integrates Minecraft <--> TNB

#### What is the estimated completion date for this project? (rough estimates are acceptable)
It'll take 2 weeks at the most

#### Please include a detailed roadmap for your project below. This must include a list of all milestones where payments will be awarded upon completion of each.

There's only one milestone and it'll take up to 2 weeks.

#### What is your Discord username?
Pixel#5000

# Milestone Details

#### Milestone Description
Milestone 1 - implement the following:

/tnb deposit
/tnb tip <PlayerName> [amount]
/tnb withdraw <address> <amount>

#### Requested Payment Upon Completion
10,000 TNBC - bear in mind this includes good code and tests- we're dealing with money!

| # | Objective           | Estimated Value (coins)   |
| - | ------------------- | ------------------------- |
| 1 | Depositing TNB | 3333 |
| 2 | Withdrawing TNB | 3333 |
| 3 | Tipping other players | 3333 |

#### Project Wallet
add7f7dc1f719179530f7e6c39cb70b2dd671119b63159e1383b190912b8e1c7

**Comment by AbhayAysola** [comment]
This is a pretty cool idea! Can I get some implementation details if you have it planned? Also, are you sure that the Minecraft EULA permits something like this. I think it would be wise to read through once and do some research on it.

**Comment by mrbusysky** [comment]
Make sure to read minecrafts TOS before you do this because users can't use crypto or fiat to buy items land or anything like that or minecraft can ban people from their network.

**Comment by manishram** [comment]
Hey, @Kycermann we really liked your idea. Your proposal has been passed to govs. for final review.

**Comment by angle943** [comment]
1. is `/tnb tip [amount]` a tip to a specific player? If so, shouldn't we write their username, ex: `/tnb tip [amount] [username]`?

2. how do you register your tnb account with the minecraft account?

3. How will `/tnb rain` work exactly?

4. when you say `tip players (without fees)`, what fee are you talking about exactly? You don't mean the validation fees, right?

**Comment by Kycermann** [comment]
@angle943

1. Yep
2. You don't- a TNB balance is tied to each Minecraft UUID (unique user ID). The actual TNB is held by the plugin in one address.
3. Distribute [amount] evenly (or randomly, I'll do some empirical tests and see what works best) between online players who aren't AFK
4. I mean players can transfer balance between each other without creating TNB transactions because until you withdraw, the TNB is in an address on the server
5. When depositing, the memo content is the target user's UUID- so you can deposit into any UUID and we physically can't stop that. When sending TNB in-game, you'll only be able to send TNB to players who have joined the [current] server before.
6. The plugin will have a TNB keypair, they'll go there.
7 & 8. It's up to the server owner! Minecraft policies allow selling cosmetics.

**Comment by angle943** [comment]
Approved by the gov't for 10000

**Comment by FilterUnfiltered** [comment]
I don't think this is allowed by Minecraft TOS if these coins can be used to encourage pay-to-win, which it will. Minecraft only allows cosmetics and things that do not affect gameplay level to be added to the game. I might be wrong, and correct me if I am, but AFAIK this breaks the Minecraft server EULA

**Comment by Kycermann** [comment]
> I don't think this is allowed by Minecraft TOS if these coins can be used to encourage pay-to-win, which it will. Minecraft only allows cosmetics and things that do not affect gameplay level to be added to the game. I might be wrong, and correct me if I am, but AFAIK this breaks the Minecraft server EULA

Nope, we're all good.
